warehouse/query-core/src/main/java/datawave/query/postprocessing/tf/TFFactory.java

```diff
@@ -40,29 +43,34 @@
             
         }
         
-        return getFunction(query, contentExpansionFields, termFrequencyFields, fieldMappings, equality, evaluationFilter, sourceCopy, tfIndexOnlyFields);
+        return getFunction(query, contentExpansionFields, termFrequencyFields, fieldMappings, equality, evaluationFilter, sourceCopy, secondSource, isTld);
     }
     
     /**
      * Factory method for creating the TF function used for generating the map context.
      * 
      * @param query
+     *            the query tree
      * @param dataTypes
      * @param sourceDeepCopy
+     *            a source copy used to fetch term frequencies
+     * @param secondSource
+     *            a source copy used to fetch delayed values from the field index (negated content function terms)
      * @return
      */
     public static com.google.common.base.Function<Tuple2<Key,Document>,Tuple3<Key,Document,Map<String,Object>>> getFunction(ASTJexlScript query,
                     Set<String> contentExpansionFields, Set<String> termFrequencyFields, Multimap<String,Class<? extends Type<?>>> dataTypes,
-                    Equality equality, EventDataQueryFilter evaluationFilter, SortedKeyValueIterator<Key,Value> sourceDeepCopy, Set<String> tfIndexOnlyFields) {
+                    Equality equality, EventDataQueryFilter evaluationFilter, SortedKeyValueIterator<Key,Value> sourceDeepCopy,
+                    SortedKeyValueIterator<Key,Value> secondSource, boolean isTld) {
```

Comment:
Consider the api ergonomics here. Is it reasonable to expect the caller to know that it should `deepCopy` the source twice? Or should the caller just send `TFFactory` a source, and `TFFactory` is responsible for deepCopying it as many times as the implementation requires?  It also makes me wonder whether we should bundle all of these 'configuration' objects like `contentExpansionFields`, `termFrequencyFields`, `tfIndexOnlyFields` together into a single parent configuration object.

Reply:
Makes sense to bundle all of these into a single object that gets passed around.